_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -1066,6 +1070,126 @@ private function _process_onboarding( $data ) {
 			: join( ', ', $error );
 	}
 
+	static function get_first_sidebar() {
+		$active_sidebars = get_option( 'sidebars_widgets', array() );
+		unset( $active_sidebars[ 'wp_inactive_widgets' ], $active_sidebars[ 'array_version' ] );
+
+		if ( empty( $active_sidebars ) ) {
+			return false;
+		}
+		$active_sidebars_keys = array_keys( $active_sidebars );
+		return array_shift( $active_sidebars_keys );
+	}
+
+	static function add_business_address( $data ) {
+		$first_sidebar = self::get_first_sidebar();
+
+		if ( $first_sidebar ) {
+			$title = $data['businessAddress'][ 'name' ];
+			$address =
+					$data[ 'businessAddress' ][ 'city' ] . ' ' .
+					$data[ 'businessAddress' ][ 'state' ] . ' ' .
+					$data[ 'businessAddress' ][ 'street' ] . ' ' .
+					$data[ 'businessAddress' ][ 'zip' ];
+			$widget_options = array(
+				'title'   => $title,
+				'address' => $address,
+				'phone'   => '',
+				'hours'   => '',
+				'showmap' => false,
+				'email' => ''
+			);
+
+			if ( ! self::has_business_address_widget( $first_sidebar ) ) {
+				self::insert_widget_in_sidebar( 'widget_contact_info', $widget_options, $first_sidebar );
+			} else {
+				self::update_widget_in_sidebar( 'widget_contact_info', $widget_options, $first_sidebar );
+			}
+			wp_send_json_success( array( 'updated' => true ) );
+			die();
+		}
+		wp_send_json_success( array( 'updated' => false ) );
```

Comment:
We're not supposed to call `wp_send_json_success` here, because it'll end the request in an unexpected way. `Jetpack_Core_API_Data::update_data()` already handles this by either calling a `rest_ensure_response()` on success, or returning a `WP_Error` object on failure, so we only need to pass on the errors, if any, and success messages, if any (following the pattern that `_process_onboarding` already implements).